src/Database/Type/DateTimeType.php

```diff
@@ -295,7 +320,8 @@ public function manyToPHP(array $values, array $fields, DriverInterface $driver)
     public function marshal($value): ?DateTimeInterface
     {
         if ($value instanceof DateTimeInterface) {
-            return $value;
+            /** @var \Datetime|\DateTimeImmutable $value */
+            return $value->setTimezone($this->defaultTimezone);
```

Comment:
Should we clone here for mutable instances? This is actually modifying the expected test result.

Reply:
Done